pymilvus/decorators.py

```diff
@@ -49,22 +49,13 @@ def timeout(start_time) -> bool:
                 try:
                     return func(self, *args, **kwargs)
                 except grpc.RpcError as e:
-                    # DEADLINE_EXCEEDED means that the task wat not completed
-                    # UNAVAILABLE means that the service is not reachable currently
                     # Reference: https://grpc.github.io/grpc/python/grpc.html#grpc-status-code
-                    if e.code() != grpc.StatusCode.DEADLINE_EXCEEDED and e.code() != grpc.StatusCode.UNAVAILABLE:
-                        raise MilvusException(message=str(e)) from e
-                    if not retry_on_deadline and e.code() == grpc.StatusCode.DEADLINE_EXCEEDED:
+                    if e.code() in (grpc.StatusCode.DEADLINE_EXCEEDED, grpc.StatusCode.UNIMPLEMENTED):
```

Comment:
@jaime0815 Are you sure about adding the other 14 status code in to retriable exceptions?https://grpc.github.io/grpc/core/md_doc_statuscodes.html

To me, there're exceptions unable to retry, like INVALID_ARGUMENT, UNKNOWN, CANCELED, ABORTED, PERMISSION_DENIED, etc.

Better to limit the retriable grpc Errors instead of retrying all 14 errors.